src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -255,50 +262,67 @@ def _get_node_maps(self, cores, gpus, threads_per_proc):
     def _alloc_nompi(self, cud):
         '''
         Find a suitable set of cores and gpus *within a single node*.
+
+        Input:
+        cud: Compute Unit description. Needs to specify at least one process,thread
+        or GPU process.
+
         '''
 
-        # dig out the allocation request details
+        # Get the number of processes, threads of GPU processes this unit requests.
+        # For threads, if the entry is not specified in the description, then 
+        # assume it is one thread per process.
         requested_procs  = cud['cpu_processes']
-        threads_per_proc = cud['cpu_threads']
+        threads_per_proc = cud.get('cpu_threads',1)
         requested_gpus   = cud['gpu_processes']
 
-        if not threads_per_proc:
-            threads_per_proc = 1
+        # The total number of requested logical cores is equal to the number of
+        # processes times the number of threads per process.
 
         requested_cores = requested_procs * threads_per_proc
 
-        # we want the allocation to fit on a single node - make sure this is
-        # actually possible
+        # The allocation has to fit in a single node. If this is true contnue and
+        # try to find a possible allocation, otherwise raise a value error exception.
         if  requested_cores > self._lrms_cores_per_node or \
```

Comment:
'fit in' -> 'fit on'
'contnue' -> 'continue'

Again I am not sure if the change really adds any information, apart from spelling out `if a>b: raise ValueError`, which is very readable in the code already.